src/Compilers/Core/Portable/Syntax/SyntaxList`1.cs

```diff
@@ -472,6 +516,22 @@ public int IndexOf(TNode node)
             return -1;
         }
 
+        /// <summary>
+        /// The index of the node in this list, or -1 if the node is not within the range of elements that extends from the specified index to the last element.
+        /// </summary>
+        public int IndexOf(TNode node, int startIndex)
+        {
+            for (int index = 0; index < this.Count; index++)
+            {
+                if (EqualityComparer<TNode>.Default.Equals(this[index], node))
+                {
+                    return index;
+                }
+            }
+
+            return -1;
+        }
```

Comment:
doesn't feel like this is really necessary.  And, if it was, it could be an extension.  Also, if we update this guy, we should ad these methods to all the other syntax list kinds.  i.e. SeparatedList, TokenList, etc.